### PR TITLE
Added program status filtering

### PR DESCRIPTION
**Pull request by clintonb**
The programs API endpoint now supports filtering by status.

ECOM-7625

### CODE REVIEW

**Comment by clintonb** [comment]
@edx/ecommerce @edx/helio please review

**Review by schenedx — APPROVED** [review]
👍

File: course_discovery/apps/api/v1/tests/test_views/test_programs.py [hunk 1]
```diff
@@ -166,6 +166,20 @@ def test_filter_by_marketable(self, status, is_marketable, expected_query_count)
         self.assertEqual(list(Program.objects.marketable()), expected)
         self.assert_list_results(url, expected, expected_query_count)
 
+    def test_filter_by_status(self):
+        """ Verify the endpoint allows programs to filtered by one, or more, statuses. """
+        active = ProgramFactory(status=ProgramStatus.Active)
+        retired = ProgramFactory(status=ProgramStatus.Retired)
+
+        url = self.list_path + '?status=active'
```
Comment (vkaracic) [inline course_discovery/apps/api/v1/tests/test_views/test_programs.py:174]: Why not ``format()``?

File: course_discovery/apps/api/v1/tests/test_views/test_programs.py [hunk 2]
```diff
@@ -166,6 +166,20 @@ def test_filter_by_marketable(self, status, is_marketable, expected_query_count)
...
+        self.assert_list_results(url, [active], 8)
```
Comment (vkaracic) [inline course_discovery/apps/api/v1/tests/test_views/test_programs.py:175]: ``8`` could be a constant here.

**Review by rlucioni — APPROVED** [review]

**Pull request merged**